events/events.go

```diff
@@ -31,9 +35,83 @@ type EventListener interface {
 	HandleEvent(event Event)
 }
 
+// EventEmitter describes an interface that can be used to emit events
+type EventEmitter interface {
+	EmitEvent(Event)
+}
+
 // EventRegistrar is an object that you can register EventListeners on.
 type EventRegistrar interface {
 	RegisterListener(EventListener)
+	DeregisterListener(EventListener)
```

Comment:
I agree with the better naming, however we will not be able to change `RegisterListener` easily as it will break the public interface of ringpop which external libraries rely on for functioning.

We might want to use your proposed names here as the actual functions, and keep at least the `RegisterListener` function as a passthrough to `AddListener`.